File: 3430-count-days-without-meetings/3430-count-days-without-meetings.cs
public class Solution {
    public int CountDays(int days, int[][] meetings) {
        /// Merge all days
        Array.Sort(meetings, (a, b) => a[0].CompareTo(b[0]));    
        int[] cur = new int[2];   
        cur[0] = meetings[0][0];
        cur[1] = meetings[0][1];
        List<int[]> new_merge = new List<int[]>();

        for(int i = 0;i<meetings.Length;i++)
        {
            if (meetings[i][0] <= cur[1])
                cur[1] = Math.Max(cur[1], meetings[i][1]);
            else{
                new_merge.Add(cur.ToArray());
                cur[0] = meetings[i][0];
                cur[1] = meetings[i][1];
            }
        }        
        new_merge.Add(cur);

        int res = 0;
        foreach(var item in new_merge)
        {
            res+= item[1]-item[0]+1     ;       
        }
        
        return days-res;
    }
}

File: 3430-count-days-without-meetings/3430-count-days-without-meetings.py
class Solution:
    def countDays(self, days: int, meetings: List[List[int]]) -> int:
        meetings.sort()
        # print(meetings)
        new_arr = []
        cur_arr = meetings[0]
        # do merge
        for i in range(len(meetings)):
            s,e = meetings[i][0],meetings[i][1]
            if s <= cur_arr[1]:
                cur_arr[1] = max(cur_arr[1],e)
            else:
                new_arr.append(cur_arr)
                cur_arr = meetings[i]
        new_arr.append(cur_arr)
        print(new_arr)
        res = 0
        for s,e in new_arr:
            res += (e-s+1)
        return days-res      




